src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Date.cs

```diff
@@ -29,6 +29,11 @@ public static void TrimDateTimeOffset(Span<byte> buffer, out int bytesWritten)
                 buffer.Length == (JsonConstants.MaximumFormatDateTimeLength + 1) ||
                 buffer.Length == JsonConstants.MaximumFormatDateTimeOffsetLength);
 
+            if (buffer.Length == JsonConstants.MaximumFormatDateTimeOffsetLength)
```

Comment:
This is an unnecessary check since its only relevant in Debug mode.

**Either:**
1) Surround it within an #if DEBUG so that it doesn't show up in the release product. OR
2) Move the Debug.Assert down to line 100/101 after `byte offsetMinDigit1 = buffer[31];`, where it actually matters (my preference, but either works).

Generally, it doesn't make sense to add branches on release where the logic contained is just used on debug.